### ISSUE TITLE
Parse hook should not be used when doing --new-pr or any other -pr related command

### ISSUE THREAD

**Issue by akesandgren** [easybuilders/easybuild-framework#2600]
We do not want site specific changes into the PRs made by --new-pr

The other hooks might also cause problems and should also be avoided for -pr related commands.

**Comment by boegel** [comment]
Do you have a concrete example where this goes wrong?

I wouldn't expect that with `--new-pr` & co site-specific changes that are being injected via hooks are picked up, since the `--*-pr` options just copy the original file(s) that were provided, don't they?

What's the problem with `--inject-checksums` exactly? Is the stuff you're doing in the hooks messing it up (or the other way around)?

There are a couple of ways to avoid that the hooks are picked up:

* in `main.py`, you could make the call to `load_hooks` conditional, i.e. not do it when certain options are enabled; the fallback value for `hooks` should be `{}`
* in the `postprocess` method `options.py`, you can hard overwrite the value for the `--hooks` option to be empty when `--new-pr` & co are used (there's a similar example where `--backup-modules` is auto-enabled when `--module-only` or `--skip` are used)

I'm not sure where we should hardcode that though, maybe it should be optional whether or not the hooks are enabled for `--new-pr` & co?
I guess whether or not you want to depends on what you do in the hooks... I can imagine a use case where hooks are used to try and clean up the easyconfigs (remove site-specific stuff) when `--new-pr` is used for example...

Another option is to make your hook implementations "abort" when `--new-pr` (or any other option that gets in the way) is used, you should be able to detect that (and if not, it's easy to fix).

**Comment by akesandgren** [comment]
I did a --new-pr for something that used a already existing foss toolchain, and it picked up our extra dependency of PMIx for OpenMPI and failed (luckily)

**Comment by boegel** [comment]
@akesandgren How did it fail exactly? Were you combining `--new-pr` with `--robot`, and did it fail to find an easyconfig for `PMIx`?

I think if it wouldn't have failed, the easyconfig file that got contributed wouldn't have had `PMIx` as dependency. I probably would have included the `PMIx` easyconfig file if it found it though, so that's not good.

**Comment by akesandgren** [comment]
Example:
Added a dummy UCY dependency to OpenMPI in our parse hook, and created the corresponding easyconfig.
eb --new-pr GROMACS-2018.2-foss-2018b-dummy.eb -Dr --disable-cleanup-tmpdir
* overview of changes:
 .../g/GROMACS/GROMACS-2018.2-foss-2018b-dummy.eb   | 43 ++++++++++++++++++++++
 .../easyconfigs/u/UCY/UCY-1.3.1-GCCcore-7.3.0.eb   | 42 +++++++++++++++++++++

**Comment by akesandgren** [comment]
However, for the two (real) extra deps our OpenMPI have, PMIx and UCX which both already are in develop, they wheren't added to the PR

**Comment by akesandgren** [comment]
But mainly, if a dependency added through parse_hook is doesn't exist or has problems, the --new-pr will fail during dependency checking or easyconfig processing. Which might be confusing.

**Comment by boegel** [comment]
Yeah, OK, so that indeed warrants a way to disable the hooks when `--new-pr` is used, but since there also are use cases where keeping them enabled makes sense (and the fact that they're currently enabled by default), we're looking at an opt-out config setting for this (or making sure you can opt-out in the hook implementations themselves).

**Comment by akesandgren** [comment]
And for inject-checksum, if I add a patch in the parse_hook the checksum for the patch gets added but the patch doesn't.
